feature/PPM-5 Integration - Client BML APIs

itayx opened the pull request:
Added small fixes for the smart steering BML client API
- In client_set_client changed incoming type from INT_ARG to STRING_ARG.
- Added int conversion in the client_set_client LOG print.
- Added tlvf::mac_to_string in the client_get_client LOG print.

itayx committed:
bml: BML_CLIENT: change sta_mac type to unit8_t

The type of the mac address parameter is incorrect.

Changed sta_mac from char[6] to uint8_t[6].

Signed-off-by: Itay Elenzweig <itayx.elenzweig@intel.com>

itayx committed:
beerocks_cli_bml: Fix client APIs

- In client_set_client changed incoming type from INT_ARG to STRING_ARG.
Since we are using optional parameters, we need to retrieve them from
the CLI as strings, then convert them to integers.

- Added int conversion in the client_set_client LOG print.
To print the incoming values in client_set_client, need to convert them
to type int, otherwise they will be printed as Char

- Added tlvf::mac_to_string in the client_get_client LOG print.
The received mac address needs to be converted to a readable format to
be printed.

Signed-off-by: Itay Elenzweig <itayx.elenzweig@intel.com>

adam1985d commented on controller/src/beerocks/bml/bml_defs.h:
```diff
@@ -599,7 +599,7 @@ struct BML_CLIENT_CONFIG {
 struct BML_CLIENT {
     // Client MAC.
-    char sta_mac[BML_MAC_ADDR_LEN];
+    uint8_t sta_mac[BML_MAC_ADDR_LEN];
```
commit **title** should be:
`bml: BML_CLIENT: change sta_mac type to unit8_t`
or
`bml: change sta_mac type in BML_CLIENT to unit8_t`

1. bml not bcl
2. current title doesn't say anything or doesn't say enough

itayx replied:
Changed

adam1985d commented on controller/src/beerocks/cli/beerocks_cli_bml.cpp:
```diff
@@ -563,8 +563,8 @@ void cli_bml::setFunctionsMapAndArray()
         " stay_on_selected_device - 1 for true, 0 for false or (default) -1 for not configured",
-        static_cast<pFunction>(&cli_bml::client_set_client_caller), 2, 4, STRING_ARG, INT_ARG,
-        INT_ARG, INT_ARG);
+        static_cast<pFunction>(&cli_bml::client_set_client_caller), 2, 4, STRING_ARG, STRING_ARG,
```
1. commit message:
- change `beerocks_cli` to `beerocks_cli_bml`
- please explain on each change why the changes are needed

itayx replied:
Changed

adam1985d commented on controller/src/beerocks/cli/beerocks_cli_bml.cpp:
```diff
@@ -2230,9 +2230,10 @@ int cli_bml::client_set_client(const std::string &sta_mac, int8_t selected_bands
+              << "  selected_bands: " << std::to_string(selected_bands) << std::endl
+              << "  stay_on_initial_radio: " << std::to_string(stay_on_initial_radio) << std::endl
+              << "  stay_on_selected_device: " << std::to_string(stay_on_selected_device)
+              << std::endl;
```
```suggestion
              << "  selected_bands: " << int(selected_bands) << std::endl
              << "  stay_on_initial_radio: " << int(stay_on_initial_radio) << std::endl
              << "  stay_on_selected_device: " << int(stay_on_selected_device)
              << std::endl;
```

itayx replied:
Changed

adam1985d reviewed (approved):
looks good - gave small suggestions to the commit messages and other, but simple fixes so approving

itayx committed:
controller: bml: fix return values

Changed returning error values in the code to a negative value as
expected by the BML.

PPM-5.

Signed-off-by: Itay Elenzweig <itayx.elenzweig@intel.com>

itayx committed:
controller: bml: refactor client_get_client

Refactored the response handling for the "get_client" implementation.
Fixed broken return on error in the "get_client" implementation.

PPM-5.
Signed-off-by: Itay Elenzweig <itayx.elenzweig@intel.com>

itayx committed:
controller: bml: verify client_config parameters

As part of the persistent DB, some configuration parameters can be set
to "NOT_CONFIGURED" meaning they will not be stored persistently.

Verify that the given client configuration parameters are configured
before sending the change request.

Signed-off-by: Itay Elenzweig <itayx.elenzweig@intel.com>
